test/PoolManager.t.sol

```diff
@@ -982,19 +1002,48 @@ contract PoolManagerTest is Test, Deployers, GasSnapshot {
         manager.setProtocolFee(key);
     }
 
-    function test_collectProtocolFees_initializesWithProtocolFeeIfCalled() public {
-        uint16 protocolFee = 1028; // 00000100 00000100
-
-        // sets the upper 12 bits
-        feeController.setSwapFeeForPool(uninitializedKey.toId(), uint16(protocolFee));
+    function test_collectProtocolFees_ERC20_returnsCorrectFeesWithParameters(uint256 swapAmount, uint256 balanceToClaim)
+        public
+    {
+        (, uint256 amount1) = getMaxAmountInForPool(Deployers.LIQ_PARAMS, key);
+        // lower bound for precision purposes
+        swapAmount = uint256(bound(swapAmount, 100, amount1));
 
-        initializeRouter.initialize(uninitializedKey, SQRT_RATIO_1_1, ZERO_BYTES);
-        (Pool.Slot0 memory slot0,,,) = manager.pools(uninitializedKey.toId());
+        uint16 protocolFee = 1028; //  swap fee0 = 4, fee1 = 4
+        address protocolFeeRecipient = address(1);
+        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
+        manager.setProtocolFee(key);
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
         assertEq(slot0.protocolFee, protocolFee);
+
+        // token 1 for 0
+        IPoolManager.SwapParams memory swapParams =
+            IPoolManager.SwapParams(false, SafeCast.toInt256(swapAmount), TickMath.MAX_SQRT_RATIO - 1);
+        swapRouter.swap(key, swapParams, PoolSwapTest.TestSettings(true, true, false), ZERO_BYTES);
+
+        uint256 feesAccrued = manager.balanceOf(address(feeController), currency1.toId());
+        console2.log(feesAccrued);
+
+        vm.prank(address(feeController));
+        if (balanceToClaim == 0 || balanceToClaim == feesAccrued) {
+            // Expect that they claimed their entire balance
+            manager.collectProtocolFees(protocolFeeRecipient, currency1.toId(), balanceToClaim);
+            assertEq(currency1.balanceOf(protocolFeeRecipient), feesAccrued);
+            assertEq(manager.balanceOf(address(feeController), currency0.toId()), 0);
```

Comment:
```suggestion
            assertEq(manager.balanceOf(address(feeController), currency1.toId()), 0);
```

i think this needs to be for currency1 to check that their 6909 balance is 0 (because fees were accrued in currency1)

Reply:
Ah yes great catch